pkg/services/ngalert/state/state.go

```diff
@@ -75,6 +76,38 @@ type State struct {
 	EvaluationDuration   time.Duration
 }
 
+// Copy creates a shallow copy of the State except for labels and annotations.
+func (a *State) Copy() *State {
+	// Deep copy annotations and labels
+	annotationsCopy := make(map[string]string, len(a.Annotations))
+	maps.Copy(annotationsCopy, a.Annotations)
+	labelsCopy := make(data.Labels, len(a.Labels))
+	maps.Copy(labelsCopy, a.Labels)
+	valuesCopy := make(map[string]float64, len(a.Values))
+	maps.Copy(valuesCopy, a.Values)
+	return &State{
+		OrgID:                a.OrgID,
+		AlertRuleUID:         a.AlertRuleUID,
+		CacheID:              a.CacheID,
+		State:                a.State,
+		StateReason:          a.StateReason,
+		ResultFingerprint:    a.ResultFingerprint,
+		LatestResult:         a.LatestResult,
+		Error:                a.Error,
+		Image:                a.Image,
+		Annotations:          annotationsCopy,
+		Labels:               labelsCopy,
+		Values:               a.Values,
```

Comment:
I think it's a typo, it should be `valuesCopy`

Reply:
no, actually valuesCopy is a leftover I forgot to remove :) this method is intended to be a shallow copy state except for labels and annotations which are mutated downstream